api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -87,6 +87,13 @@ message HeaderMutationRules {
   google.protobuf.BoolValue disallow_is_error = 7;
 }
 
+// The HeaderForwardingRules structure specifies what headers are
+// allowed to be forwarded to the ext_proc service.
+message HeaderForwardingRules {
+  // Only allow headers that matching this regular expression to be forwarded.
+  repeated type.matcher.v3.RegexMatcher headers_to_forward = 1;
```

Comment:
I know that this is more convenient but what is the perf tax we are paying? For every header, we will have to match each one of these regexes to determine if it can be sent or not sent. Would it be better to start with simple exact match and then expand to prefix/regex as time progresses? This way, the common case of exact match is a reverse lookup from headers_to_forward to the header map as opposed to the current model where we have to do a O(N) pass of the header_map and run each header by a compound regex

Reply:
I think we could do a `repeated StringMatcher`. That way, when every entry is an exact match you can build a hash table, otherwise you can do the O(N) evaluation (or tries if you are really fancy). By using StringMatcher you leave room to grow without API change.

Reply:
+1 for StringMatcher. Actually there is already one that you can use directly --- [ListStringMatcher](https://github.com/envoyproxy/envoy/blob/5e7a4b5f44f398a42897f743a8665ce92fc563a3/api/envoy/type/matcher/v3/string.proto#L73).

Reply:
+1 to @tyxia suggestion of using `ListStringMatcher`. This is also used in [ext_authz](https://github.com/envoyproxy/envoy/blob/5e7a4b5f44f398a42897f743a8665ce92fc563a3/api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto#L181)

Reply:
Good idea. Changed into ListStringMatcher.

Reply:
BTW, based on the comments above, we just need to support EXACT match in the initial implementation.